cpp/src/arrow/array/array_nested.h

```diff
@@ -370,6 +370,13 @@ class ARROW_EXPORT StructArray : public Array {
   /// \param[in] pool The pool to allocate null bitmaps from, if necessary
   Result<ArrayVector> Flatten(MemoryPool* pool = default_memory_pool()) const;
 
+  /// \brief Get one of the child arrays, adjusting the null bitmap if necessary.
+  ///
+  /// \param[in] index Which child array to get
+  /// \param[in] pool The pool to allocate null bitmaps from, if necessary
+  Result<std::shared_ptr<Array>> Flatten(int index,
```

Comment:
I find `Flatten` a bit confusing, but I understand there's precedent.

Reply:
We could call it `MakeFlattenedChild` or `GetFlattenedChild` or something if that helps? I agree Flatten isn't the best name when it comes to a single child array.

Reply:
Updated this and fixed the copy above to use std::copy.